blocks/editable/index.js

```diff
@@ -123,6 +126,38 @@ export default class Editable extends Component {
 		}
 	}
 
+	onPastePostProcess( event ) {
+		const childNodes = Array.from( event.node.childNodes );
+		const isBlockDelimiter = ( node ) =>
+			node.nodeType === 8 && /^ wp:/.test( node.nodeValue );
+		const isDoubleBR = ( node ) =>
+			node.nodeName === 'BR' && node.previousSibling && node.previousSibling.nodeName === 'BR';
+		const isBlockPart = ( node ) =>
+			isDoubleBR( node ) || this.editor.dom.isBlock( node );
+
+		// If there's no `onSplit` prop, content will later be converted to
+		// inline content.
+		if ( this.props.onSplit ) {
+			let blocks = [];
+
+			// Internal paste, so parse.
+			if ( childNodes.some( isBlockDelimiter ) ) {
+				blocks = parse( event.node.innerHTML.replace( /<meta[^>]+>/, '' ) );
+			// External paste with block level content, so attempt to assign
+			// blocks.
+			} else if ( childNodes.some( isBlockPart ) ) {
+				blocks = pasteHandler( childNodes );
+			}
```

Comment:
Actually, I think part of this should be in Editable because it needs to decide what content it con handle itself (inline), and then pass the rest to the paste handler. This can probably be improved though. Maybe we should only do the black check and leave out the comment check.